cakechat/dialog_model/model_utils.py

```diff
@@ -75,7 +77,7 @@ def transform_contexts_to_token_ids(tokenized_contexts,
             if add_start_end:
                 utterance = [SPECIAL_TOKENS.START_TOKEN] + utterance + [SPECIAL_TOKENS.EOS_TOKEN]
 
-            for token_idx, token in enumerate(utterance[:max_line_len]):
+            for token_idx, token in enumerate(islice(utterance, max_line_len)):
```

Comment:
Why do you need this? Utterance is pretty short so you can just slice it in a normal way